src/components/MultiSelect/helpers/normalizeValue.js

```diff
@@ -0,0 +1,18 @@
+export default function normalizeValue(value) {
+    if (!value || typeof value !== 'object') {
+        return [];
+    }
+    if (Array.isArray(value)) {
+        if (value.length === 0) {
+            return [];
+        }
+        return value.map(v => {
+            const newValue = v;
+            delete newValue.icon;
+            return newValue;
+        });
+    }
+    const newValue = value;
+    delete newValue.icon;
```

Comment:
same here you are mutating the original value object, which is not a good practice